src/runtime-rs/crates/runtimes/src/manager.rs

```diff
@@ -434,14 +451,64 @@ impl RuntimeHandlerManager {
 
             Ok(TaskResponse::CreateContainer(shim_pid))
         } else {
-            self.handler_request(req)
+            self.handler_task_request(req)
                 .await
                 .context("handler TaskRequest")
         }
     }
 
+    pub async fn handler_sandbox_request(&self, req: SandboxRequest) -> Result<SandboxResponse> {
+        let instance = self
+            .get_runtime_instance()
+            .await
+            .context("get runtime instance")?;
+        let sandbox = instance.sandbox.clone();
+
+        match req {
+            SandboxRequest::CreateSandbox(req) => Err(anyhow!("Unreachable request {:?}", req)),
+            SandboxRequest::StartSandbox(_) => {
+                sandbox.start().await.context("start sandbox")?;
```

Comment:
`sandbox.start()` appears to be called as part of `SandboxRequest::CreateSandbox` handling already (`handler_sandbox_message()` calls `sandbox_init_runtime_instance()` which in turn calls `RuntimeHandlerManagerInner::try_init()` and `start_runtime_handler()` where a call to `Sandbox::start()` happens).  Is that expected and intentional?

Reply:
Hi @pmores 

Here the calling of sandbox.start() is intentional. Since we need to support both of the sandboxapi and the legacy of pod sandbox api, and for the legacy pod sandbox api,  the sandbox would be created and started in the task_init_runtime_instance -> { RuntimeHandlerManagerInner::try_init() -> start_runtime_handler() } call chain, while for the new sandbox api, the create sandbox was called in the sandbox_init_runtime_instance, and the start sandbox was called in SandboxRequest::StartSandbox(_), which was here.

Reply:
Thanks @lifupan , I understand the need to support the legacy pod sandbox api as well, my question is solely within the scope of the new sandbox api support.  My confusion is that the handling of the `SandboxRequest::CreateSandbox` apparently starts the sandbox already (via the prevously mentioned call chain `RuntimeHandlerManager::handler_sandbox_message()` -> `RuntimeHandlerManager::sandbox_init_runtime_instance()` -> `RuntimeHandlerManagerInner::try_init()` -> `RuntimeHandlerManagerInner::start_runtime_handler()` -> `Sandbox::start()`).

If that's the case then creating the sandbox in the new sandbox api (`SandboxRequest::CreateSandbox` ) seems to not just create a sandbox but start it as well which I'm not sure if it's right.  Also, the subsequent `SandboxRequest::StartSandbox` request seems to start the sandbox again which would also make me nervous since `Sandbox::start()` doesn't look idempotent.

Reply:
Hi @pmores 

I know what you mean now. You are right, the start sandbox in the try_init() in redundant. Now I had moved it to task_init_runtime_instance now.  Thanks.

Reply:
Thanks @lifupan , now it looks much better!